### PR TITLE
MAT-8010: Convert editor to npm module

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-0000](https://jira.cms.gov/browse/MAT-8010)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/types/madie-madie-editor.d.ts [hunk 1]
```diff
@@ -88,6 +87,7 @@ declare module "@madie/madie-editor" {
   export function isUsingEmpty(editorVal: string): boolean;
 
   export const MadieEditor: FC<{
```
Comment (RohitKandimalla) [inline src/types/madie-madie-editor.d.ts:89]: Do we still need to explicitly defines types for imported modules ?

File: package.json [hunk 2]
```diff
@@ -28,7 +28,8 @@
     "@babel/preset-typescript": "^7.14.5",
     "@babel/runtime": "^7.14.6",
     "@madie/madie-design-system": "^1.2.92",
-    "@madie/madie-models": "^1.4.62",
+    "@madie/madie-models": "1.4.62",
+    "@madie/madie-editor": "file:../madie-editor/madie-madie-editor-0.0.7.tgz",
```
Comment (RohitKandimalla) [inline package.json]: Pinningit, so that we revert this before merging

Comment (mcmcphillips) [reply package.json]: updated to 1.0

File: src/types/madie-madie-editor.d.ts (hunk 1 above)
Comment (mcmcphillips) [reply src/types/madie-madie-editor.d.ts:89]: Yes, or it will crash

**Review by RohitKandimalla — APPROVED** [review]
